src/core/utils/compatibility.js

```diff
@@ -173,8 +173,19 @@ export function getClientCompatibility({
     downloadUrl = FACEBOOK_CONTAINER_DOWNLOAD_URL;
   }
 
+  let compatible = agent.compatible && supportsClientApp;
+
+  const { browser } = userAgentInfo;
```

Comment:
I think it would be nice to wrap all of this code in `if (addon.isRestartRequired === true) { ...` since otherwise it's irrelevant.